src/CattyTests/Views/Data/BrickCellVariableDataTests.swift

```diff
@@ -0,0 +1,164 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+class BrickCellVariableDataTests: XCTestCase {
+
+    func testInitWithFrameWhenObjectVariableInContainer() {
+
+        let spriteObject = SpriteObject()
+        spriteObject.name = "testObject"
+
+        let variablesContainer = VariablesContainer()
+        let userVariable1 = UserVariable(name: "testVariable1")
+        let userVariable2 = UserVariable(name: "testVariable2")
+        let userVariable3 = UserVariable(name: "testVariable3")
+        variablesContainer.addObjectVariable(userVariable1, for: spriteObject)
+        variablesContainer.addObjectVariable(userVariable2, for: spriteObject)
+        variablesContainer.programVariableList.add(userVariable3 as Any)
+
+        let project = Project()
+        project.variables = variablesContainer
+
+        spriteObject.project = project
+
+        let script = Script()
+        script.object = spriteObject
+
+        let brick = ChangeVariableBrick()
+        brick.userVariable = userVariable2
+        brick.script = script
+
+        let brickCell = ChangeVariableBrickCell()
+        brickCell.scriptOrBrick = brick
+
+        let brickCellVariableData = BrickCellVariableData(frame: CGRect(), andBrickCell: brickCell, andLineNumber: 0, andParameterNumber: 0)
+
+        XCTAssertTrue(brickCellVariableData?.currentValue == brick.userVariable.name)
+
+        var allVariableNames = [String]()
+        if let allVariables = variablesContainer.allVariables(for: spriteObject) as? [UserVariable] {
```

Comment:
This test also passes if `allVariables` returns something different than an array of `UserVariable` and `brickCellVariableData` does not contain any values. This should not be the case. Remove the conditional `if let` cast here -- I would recommend to add the asserts directly for the `UserVariables` (e.g., `XCTAssertEqual(kLocalizedNewElement, values[0])`, `XCTAssertEqual(userVariable1.name, values[1])`). This makes it more readable.